### PR TITLE
DOC: remove superscript formatting for citation callouts

### PR DESCRIPTION
**Pull request by kanderso-nrel**
- [x] I am familiar with the [contributing guidelines](https://pvlib-python.readthedocs.io/en/latest/contributing.html)
 - [ ] Adds description and name entries in the appropriate "what's new" file in [`docs/sphinx/source/whatsnew`](https://github.com/pvlib/pvlib-python/tree/master/docs/sphinx/source/whatsnew) for all changes. Includes link to the GitHub Issue with `` :issue:`num` `` or this Pull Request with `` :pull:`num` ``. Includes contributor name and/or GitHub username (link with `` :ghuser:`user` ``).
 - [x] Pull request is nearly complete and ready for detailed review.
 - [ ] Maintainer: Appropriate GitHub Labels and Milestone are assigned to the Pull Request and linked Issue.

Prompted by Cliff's side comment here about removing superscripts here: https://github.com/pvlib/pvlib-python/issues/837#issuecomment-572667478

Ideally that formatting would be configurable in the theme but unless I missed it in [the docs](https://sphinx-rtd-theme.readthedocs.io/en/stable/configuring.html), I don't think the sphinx RTD theme has that option.  This PR just adds some custom CSS to override the superscript formatting.  Example [here](https://external-builds.readthedocs.io/html/pvlib-python/855/generated/pvlib.iam.ashrae.html#pvlib.iam.ashrae)

Understandable if folks don't want to maintain a bunch of custom CSS patches, but it was easy to fix so I figured might as well submit it.

### CODE REVIEW

File: docs/sphinx/source/conf.py [hunk 1]
```diff
@@ -222,6 +222,7 @@ def __getattr__(cls, name):
 # A workaround for the responsive tables always having annoying scrollbars.
```
Comment (wholmgren) [inline docs/sphinx/source/conf.py]: this comment should be updated

**Comment by cwhanse** [comment]
I like how the citation callouts appear with this custom feature. SciPy's docs use [1] for citations, and they aren't displayed using superscript. I think [this ](https://github.com/scipy/scipy-sphinx-theme-v2/blob/f7e0573f34388af8d967ac14b1bf486878fea8a4/scipy_sphinx_theme/static/css/theme.css#L6987) is where the citation callout style is defined (searched for footnote-reference).

**Comment by kanderso-nrel** [comment]
@cwhanse I'm not sure what that sphinx repo is, but I don't think it's the theme for the main scipy docs.  Maybe a test repo for numpy docs specifically?

As far as I can tell, the CSS class `footnote-reference` is a custom class defined in the RTD sphinx theme and since the scipy theme inherits straight from the basic theme, `footnote-reference` doesn't exist there.  SciPy does have two CSS classes applied to the callouts (`reference` and `internal`) but neither one actually adds any CSS rules, so the callouts end up with only normal link formatting over the normal body text formatting:

image

**Comment by cwhanse** [comment]
> @cwhanse I'm not sure what that sphinx repo is, but I don't think it's the theme for the main scipy docs.  Maybe a test repo for numpy docs specifically?
Maybe, or a suggested replacement. I was curious how SciPy did it, and wish I had more time to figure it out. In any case, pvlib-python docs formatting is improved with your addition.

**Pull request merged**